internal/component/remote/aws/secrets_manager/watcher.go

```diff
@@ -0,0 +1,109 @@
+package secrets_manager
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/secretsmanager"
+)
+
+type watcher struct {
+	mut           sync.Mutex
+	secretId      string
+	secretVersion string
+	output        chan result
+	pollTicker    *time.Ticker
+	client        *secretsmanager.Client
+}
+
+type result struct {
+	secret        string
+	secretId      string
+	secretVersion string
+	err           error
+}
+
+func (r result) Validate() bool {
+	return r.secretId != "" && r.secretVersion != "" && r.err == nil
+}
+
+func newWatcher(
+	secretId, secretVersion string,
+	out chan result,
+	frequency time.Duration,
+	client *secretsmanager.Client,
+) *watcher {
+	w := &watcher{
+		secretId:      secretId,
+		secretVersion: secretVersion,
+		output:        out,
+		client:        client,
+	}
+
+	if frequency > 0 {
+		w.pollTicker = time.NewTicker(frequency)
+	}
+	return w
+}
+
+func (w *watcher) updateValues(secretId, secretVersion string, frequency time.Duration, client *secretsmanager.Client) {
+	w.mut.Lock()
+	defer w.mut.Unlock()
+	w.secretId = secretId
+	w.secretVersion = secretVersion
+	if w.pollTicker != nil && frequency > 0 {
+		w.pollTicker.Reset(frequency)
+	}
+	w.client = client
+}
+
+func (w *watcher) run(ctx context.Context) {
+	w.getSecretAsynchronously(ctx)
+	defer w.pollTicker.Stop()
+	for {
+		select {
+		case <-w.pollTicker.C:
+			w.getSecretAsynchronously(ctx)
+		case <-ctx.Done():
+			return
+		}
+	}
+}
+
+func (w *watcher) getSecretAsynchronously(ctx context.Context) {
```

Comment:
nit: technically this method gets the secret synchronously. A more fitting name might be `getAndSendSecret`, as the method gets the secret and then writes the result to a channel.